src/dataflow-types/src/types.rs

```diff
@@ -1400,6 +1408,31 @@ pub mod sources {
         pub struct RegexEncoding {
             pub regex: mz_repr::adt::regex::Regex,
         }
+
+        impl Arbitrary for RegexEncoding {
+            type Strategy = BoxedStrategy<Self>;
+            type Parameters = ();
+
+            fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+                any_regex()
+                    .prop_map(|regex| RegexEncoding { regex })
+                    .boxed()
+            }
+        }
```

Comment:
Because `Regex` doesn't implement `Arbitrary`! I'm assuming it was an intentional choice to just provide `any_regex` as a generator instead of an `Arbitrary` implementation, at least.